Assets/Branch/Branch.cs

```diff
@@ -68,7 +68,7 @@ public class Branch : MonoBehaviour {
      * Get the referring parameters from the initial install.
      */
     public static Dictionary<string, object> getFirstReferringParams() {
-        string firstReferringParamsString = _getFirstReferringParams();
+        string firstReferringParamsString = Marshal.PtrToStringAnsi(_getFirstReferringParams());
```

Comment:
Hi,
As far as I tested, I can get custom parameter strings that way. Actually there is no built in method to get those bytes as utf8 string. Probably reading the byte array and then convert it back to a utf8 string could work.

BTW I just made a quick test trying to add some special characters to a parameter and I found out that the create url methods make the app crash too, the url is not created, so I was not able to read back its parameters. I will try to research more deeply this issue.